Library/Homebrew/tap.rb

```diff
@@ -694,6 +696,16 @@ def formula_renames
     end
   end
 
+  # Hash with tap cask renames.
+  sig { returns(T::Hash[String, String]) }
+  def cask_renames
+    @cask_renames ||= if (rename_file = path/HOMEBREW_TAP_CASK_RENAMES_FILE).file?
```

Comment:
This won't work with the API so may be worth skipping?

Reply:
Skip based on what?

Reply:
I mean: I'm not sure it's worth implementing this logic for only on disk taps when we're primarily going to be using it for homebrew-cask to begin with which won't work like this because there's no `path` present on disk.

Reply:
`fomula_renames` does the same thing which will also need to be changed in the same way to support the API. Also, I can only test this with a local tap as-is.

Reply:
Yeh, it needs implemented for formulae too. My point is: we need to implement this in the cask API before we can roll out any renames to users so: totally fine to test this as-is for now but will need API support before it can be used in any taps.

Reply:
> before it can be used in any taps

I mean it can be used like this for third-party taps which aren't using the API (homebrew/cask-versions is not using the API, so we could actually already use it there and migrate the casks with the new name). Just because we will mainly be using it in homebrew/cask doesn't mean third-party taps shouldn't have support for this.

Even with API support, `Tap#cask_renames` will still need to exist but get the rename information from the API JSON instead I assume.

Reply:
> Just because we will mainly be using it in homebrew/cask doesn't mean third-party taps shouldn't have support for this.
> Even with API support, Tap#cask_renames will still need to exist but get the rename information from the API JSON instead I assume.

agreed 👍🏻